libsoftwarecontainer/src/container.cpp

```diff
@@ -705,6 +710,50 @@ ReturnCode Container::setEnvironmentVariable(const std::string &var, const std::
     return ReturnCode::SUCCESS;
 }
 
+ReturnCode Container::freeze()
+{
+    if (m_state < ContainerState::STARTED) {
+        log_error() << "Container is not started yet";
+        return ReturnCode::FAILURE;
+    }
+
+    if (m_state == ContainerState::FROZEN) {
+        log_error() << "Container is already frozen";
+        return ReturnCode::FAILURE;
+    }
+
+    log_debug() << "Freezing container";
+    bool retval = m_container->freeze(m_container);
+
+    if (retval) {
+        m_state = ContainerState::FROZEN;
+        return ReturnCode::SUCCESS;
+    }
+
+    log_error() << "Could not freeze the container";
+    return ReturnCode::FAILURE;
+}
+
+ReturnCode Container::thaw()
+{
+    if (m_state < ContainerState::FROZEN) {
+        log_error() << "Container is not frozen";
+        return ReturnCode::FAILURE;
+    }
+
+    log_debug() << "Thawing container";
+    bool retval = m_container->unfreeze(m_container);
+
+    if (retval) {
+        // TODO: Is this really the case?
```

Comment:
I mean does a thawed container automatically become running. Docs does not seem to have a good answer for that.